edx_oauth_client/pipeline.py

```diff
@@ -99,11 +99,16 @@ def ensure_user_information(
                 r = session.get(user_info_url.format(user_data.get(settings.FEATURES['DRUPAL_ID_KEY'])))
                 api_data = r.ok and r.json() or {}
                 full_name =  (api_data.get('field_full_name', {}) or {}).get('und', [{}])[0].get('value', '')
+                fname = (api_data.get('field_first_name', {}) or {}).get('und', [{}])[0].get('value', '')
+                lname = (api_data.get('field_last_name', {}) or {}).get('und', [{}])[0].get('value', '')
                 gender = (api_data.get('field_gender', {}) or {}).get('und', [{}])[0].get('value')
                 log.info('Get gender %s for user %s', gender, user_data['email'])
                 gender = gender and gender[0].lower() or 'o'
-                full_name_list = full_name.split()
-                fname, lname = full_name_list and (full_name_list[0], ' '.join(full_name_list[1:])) or ('', '')
+                if not (fname and lname) and full_name:
+                    full_name_list = full_name.split()
+                    tmp_fname, tmp_lname = full_name_list and (full_name_list[0], ' '.join(full_name_list[1:])) or ('', '')
```

Comment:
On which case or statement will be True?

Reply:
It will be True if:
```
 full_name != ""
 and
fname = "" or lname = "" or (fname = "" and lname = "")
```
In boolean:
```
full_name != False
and
fname = False or lname = False or (fname = False and lname = False)
```